documentation/src/main/asciidoc/public/reference/_search-dsl-projection.adoc

```diff
@@ -1515,6 +1514,18 @@ include::{sourcedirJava17}/org/hibernate/search/documentation/search/projection/
 ----
 ====
 
+[[search-dsl-projection-highlight-limitations]]
+=== Highlight limitations
+
+When it comes to highlighting of fields located in <<mapping-indexedembedded-structure-nested,nested>>
+or <<mapping-indexedembedded-structure-flattened,flattened>> objects, Hibernate Search will not allow it in most scenarios right now.
+Trying to do so will result in an exception being thrown.
+The only exception to this is highlighting a field of a <<mapping-indexedembedded-structure-flattened,flattened>> object
+that is a property of a root entity, i.e. the one having a path `[flattenedObjectPropertyName].[propertyName]`.
```

Comment:
Am I really bad at reading, or is this saying "it's not supported on nested fields nor on flattened fields, with the exceptions of flattened fields"... ???

Reply:
How is that for a "better documentation" 😆 ohh 🙈 sorry, it's probably best to show it with the test and then we'll see how to update it...
sooo there is this case:

https://github.com/hibernate/hibernate-search/blob/169c8ad83ea8704346a9166a7e143fac2fb7937c/integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/HighlightProjectionTypeIndependentIT.java#L188-L194

here it doesn't matter what's the structure of `object` in `object + ".string"`, since because we are trying to add a highlight inside a `f.object( object )`.

There's also this case:

https://github.com/hibernate/hibernate-search/blob/169c8ad83ea8704346a9166a7e143fac2fb7937c/integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/AbstractHighlighterIT.java#L627-L632

where it works fine ... 🙈

Reply:
The first case uses a multi-valued object projection, which implies reconstituting the structure of the object field.

While this kind of reconstitution is restricted to nested fields on Lucene, it can work with a flattened field on Elasticsearch since Elasticsearch can use the `_source` JSON to recover the structure of an object field.

However, for highlighting, we just can't apply this "hack": we don't rely on `_source`, or at least not directly.

In the second example, you are not trying to reconstitute the structure of the object field with an object projection... so it works.